### ISSUE TITLE
Model::find memory problem in Phalcon 3.0 and PHP 7

### ISSUE THREAD

**Issue by dimak08** [phalcon/cphalcon#12144]
The cyclic Model::find()/Model::findFirst() call produce a memory problem.

Test PHP code:
```php
$loader = new \Phalcon\Loader();
$di = new \Phalcon\Di\FactoryDefault();
$di->set('db', function () use ($di) {
        $connection = new \Phalcon\Db\Adapter\Pdo\Mysql([
        /**/
    ]);
    return $connection;
}, true);

function bytes($size)
{
    $units = [' B', ' KB', ' MB', ' GB', ' TB'];
    for ($i = 0; $size >= 1024 && $i < 4; $i++)
        $size /= 1024;
    return round($size, 2) . $units[$i];
}

class Country extends \Phalcon\Mvc\Model
{
    public $id;
    public $name;
    public $flagpic;
    public $domain;
    public function getSource()
    {
        return 'countries';
    }
}
$counter = 0;
while (true) {
    $model = Country::findFirst();
    if ($counter % 1000 == 0) {
        echo bytes(memory_get_usage()) . "\n";
    }
    if ($counter > 10000) {
        die();
    }
    $counter++;
}
```

### PHP 7.0
* Phalcon version: 3.0.0
* PHP Version: 7.0.9
* Operating System: Ubuntu 16.04
* Installation type: Compiling from source (branch 3.0.x commit 649f35c9b7225b8110fd8d1e27d32434e576e058)
* Server: run in console

result: 
```
389.09 KB
3.45 MB
6.52 MB
9.62 MB
12.65 MB
15.82 MB
18.85 MB
21.89 MB
24.93 MB
28.21 MB
31.25 MB
```

### PHP 5.5
* Phalcon version: 3.0.0
* PHP Version: 5.5.9
* Operating System: Ubuntu 14.04
* Installation type: apt repository
* Server: run in console

result: 
```
301.1 KB
301.26 KB
301.26 KB
301.26 KB
301.26 KB
301.26 KB
301.26 KB
301.26 KB
301.26 KB
301.26 KB
301.26 KB
```

Valgrind log is empty.

**Comment by Jurigag** [comment]
This is not memory leak as it is. If it would be then valgrind would show anything :)

**Comment by Jurigag** [comment]
Okay. There is some problem with phql internal cache(but it shouldn't work i think on php 7, maybe make it working properly @andresgutierrez ?) After adding after findfirst:

```
$di->get('modelsManager')->__destruct();
```

It works fine.

**Comment by Jurigag** [comment]
I guess it's fixed on 3.0.1 @andresgutierrez ? Can you check it @dimak08

**Comment by dimak08** [comment]
@Jurigag Thanks, problem was fixed (288161da4009c06b19c1608f0d9b10d39d398915).
